Tests/PerformanceTests/StringTests.cs

```diff
@@ -0,0 +1,294 @@
+                        Value6 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value7 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value8 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value9 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value10 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value11 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value12 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value13 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value14 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value15 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value16 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value17 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value18 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value19 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value20 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value21 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value22 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value23 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value24 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value25 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value26 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value27 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value28 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value29 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value30 = _faker.Random.Utf16String(StringSize, StringSize),
+                        Value31 = _faker.Random.Utf16String(StringSize, StringSize),
+                    });
+                }
+            });
+
+            _robject = _realm.All<StringClass>().ElementAt(_faker.Random.Int(0, ObjectCount - 1));
+
+            _primaryKeys = new string[32];
+            for (var i = 0; i < _primaryKeys.Length; i++)
+            {
+                _primaryKeys[i] = _realm.All<StringClass>().ElementAt(_faker.Random.Int(0, ObjectCount - 1)).Value;
+            }
+
+            _newPropertyValues = new string[32];
+            for (var i = 0; i < _newPropertyValues.Length; i++)
+            {
+                var prop = typeof(StringClass).GetProperty($"Value{i}", BindingFlags.Public | BindingFlags.Instance);
+                _newPropertyValues[i] = (string)prop.GetValue(_robject);
```

Comment:
Why do we need to get the property values of the random object? Can't we just create it randomly as before?

Reply:
Here's how I understand the BenchmarkDotNet's pipeline (note that I only did some cursory reading last night, so completely possible I'm missing something):

1. It calls GlobalSetup
2. It starts calling the benchmark method until it feels it has a fairly confident measurement

Ideally, a benchmark should have no side effects, but that's kinda impossible in our case (particularly in the SetProperty case). This means, that, if we generated random values, the first time the benchmarked method is run, it would set the property from X -> Y, then all the other invocations it would be Y -> Y. Now, BenchmarkDotNet is pretty smart, so it'll do some warmup invocations and also remove outliers, so it's very unlikely that this has any effect on the measurements, but I felt it's a good idea to ensure all invocations do the same thing.

Reply:
Maybe I am missing something, but I don't think there is a difference between what you wrote and just doing
```
            for (var i = 0; i < _newPropertyValues.Length; i++)
            {
                _newPropertyValues[i] =_faker.Random.Utf16String(StringSize, StringSize);
            }
```
As you said the `SeedData` method should be called once before all the iterations start, so the elements of `_newPropertyValues` will be already defined in both cases.

Reply:
`_newPropertyValues` is just an array of in-memory values and at each benchmark invocation, we're going to call `rObject.ValueX = _newPropertyValues[x]`. The issue is that the first invocation will behave differently than subsequent invocations. Let me try to illustrate:

1. SeedData is called and let's say `rObject.Value1 = "abc"`.
2. `newPropertyValues[1] = _faker.Random...` is called, so let's say `newPropertyValues[1] = "xyz"`.
3. Benchmark.NET starts invoking the `SetPropertyValue` method.
4. First invocation, we have `rObject.Value1 = "xyz"`. This actually changes the value of `rObject.Value1` because it was previously set to `"abc"`.
5. Second invocation, we again have `rObject.Value1 = "xyz"`. This time, however, the value in the database won't be changed because it was already set to `xyz`.
6. Repeat 5. until we get a high confidence measurement.

Now, I don't think that one iteration out of many will skew the results significantly. Additionally, Benchmark.NET is smart about it and will do several warmup iterations, so it will ignore the first invocation where we mutate the underlying data. That being said, I felt it's a good idea to design the test not to rely on this knowledge and instead ensure that all invocations do the right thing - i.e. set `"abc" = "abc"`. Happy to jump on a call and chat about it if my explanation is not understandable enough 😄

Reply:
Now I think I got the point. I misunderstood one thing :)